### DISCUSSION TITLE
Could not detect Visual C++ version

### DISCUSSION THREAD

**Discussion by Prathiksha0** [OpenDDS/OpenDDS#4181] [Q&A]
After installing and trying to run the OpenDDS files for the first time, we came across this error
ERROR: Could not detect Visual C++ version, try running this script from the Visual Studio Command Prompt.
Even after trying with different versions of Visual Studio and OpenDDS, the error persists.

**Comment by iguessthislldo** [comment]
It says this if there's was an error trying to check for the Visual Studio version and architecture. It needs to be to run `cl` to do this. If you run `cl` it should say something like:

```
Microsoft (R) C/C++ Optimizing Compiler Version ... for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

usage: cl [ option... ] filename... [ /link linkoption... ]
```

If it says `cl` isn't a valid program, then it might be that [C++ support is not installed](https://learn.microsoft.com/en-us/cpp/build/vscpp-step-0-installation) or you're not running the [visual studio command prompt](https://learn.microsoft.com/en-us/visualstudio/ide/reference/command-prompt-powershell)

If you can run `cl` in the same prompt you're trying to run the configure script, then this might the same issue as #4076.

**Comment by Prathiksha0** [comment]
I'm able to run `cl`. 

Below is the error received:
`Use of uninitialized value in concatenation (.) or string at configure line 47.
ERROR: Could not detect Visual C++ version, try running this script from the Visual Studio Command Prompt.
 Stopped at configure line 618.
ERROR: configure failed with errorcode 2`

**Comment by mitza-oci** [comment]
Please run `configure` with `--verbose` and paste the output here

**Comment by Prathiksha0** [comment]
Output after running configure --verbose pasted below:
<img width="688" alt="snap" src="https://github.com/OpenDDS/OpenDDS/assets/136313282/d202a244-8228-4fc0-b07f-79fbe6b581ab">

**Comment by mitza-oci** [comment]
Which version of OpenDDS is this?  Which version/distribution of Perl?
The configure script I'm looking at shouldn't really have an "uninitialized value" error on line 47 of configure.

**Comment by Prathiksha0** [comment]
We are using **OpenDDS-3.24.1** latest version. We also tried with OpenDDS-3.18 version. For Perl we are using **Strawberry Perl**.

This is the path setting:
_System variables_:
![image](https://github.com/OpenDDS/OpenDDS/assets/136313282/d88ee579-840b-418a-ac1f-e6e7e482904a)

_User variables_:
![image](https://github.com/OpenDDS/OpenDDS/assets/136313282/de54dc77-d348-4dae-a992-1505d3bdbf7e)

**Comment by mitza-oci** [comment]
The error shown above "ERROR: Could not detect Visual C++ version, try running this ".
        "script from the Visual Studio Command Prompt"
is on line 618 in the 3.18 version, so that seems to be why I didn't see it matching.

At this point I suggest stepping through the configure script (latest release) in the perl debugger to see what's going on.  You can even try running the exact same cl command that it's running but outside of the perl script (I like using Sysinternals Process Monitor for capturing the details of what's happening when configure spawns a cl process).
